otx/algorithms/detection/utils/data.py

```diff
@@ -452,38 +454,63 @@ def adaptive_tile_params(
     Args:
         tiling_parameters (BaseTilingParameters): tiling parameters of the model
         dataset (DatasetEntity): training dataset
-        object_tile_ratio (float, optional): The desired ratio of object area and tile area. Defaults to 0.01.
+        object_tile_ratio (float, optional): The desired ratio of min object size and tile size. Defaults to 16/512=0.03.
         rule (str, optional): min or avg.  In `min` mode, tile size is computed based on the smallest object, and in
                               `avg` mode tile size is computed by averaging all the object areas. Defaults to "avg".
 
     """
     assert rule in ["min", "avg"], f"Unknown rule: {rule}"
 
-    bboxes = np.zeros((0, 4), dtype=np.float32)
+    all_sizes = np.zeros((0), dtype=np.float32)
     labels = dataset.get_labels(include_empty=False)
     domain = labels[0].domain
     max_object = 0
     for dataset_item in dataset:
         result = get_annotation_mmdet_format(dataset_item, labels, domain)
         if len(result["bboxes"]):
-            bboxes = np.concatenate((bboxes, result["bboxes"]), 0)
-            if len(result["bboxes"]) > max_object:
-                max_object = len(result["bboxes"])
-
-    areas = (bboxes[:, 2] - bboxes[:, 0]) * (bboxes[:, 3] - bboxes[:, 1])
+            bboxes = result["bboxes"]
+            sizes = 0.5 * (bboxes[:, 2] - bboxes[:, 0] + bboxes[:, 3] - bboxes[:, 1])
+            all_sizes = np.concatenate((all_sizes, sizes), 0)
+            if len(bboxes) > max_object:
+                max_object = len(bboxes)
+
+    log_sizes = np.log(all_sizes)
+    avg_log_size = np.mean(log_sizes)
+    std_log_size = np.std(log_sizes)
+    avg_size = np.exp(avg_log_size)
+    avg_3std_min_size = np.exp(avg_log_size - 3 * std_log_size)
+    avg_3std_max_size = np.exp(avg_log_size + 3 * std_log_size)
+    min_size = np.exp(np.min(log_sizes))
+    max_size = np.exp(np.max(log_sizes))
+    logger.info(f"----> [stat] log scale avg: {avg_size}")
+    logger.info(f"----> [stat] log scale avg - 3*std: {avg_3std_min_size}")
+    logger.info(f"----> [stat] log scale avg + 3*std: {avg_3std_max_size}")
+    logger.info(f"----> [stat] scale min: {min_size}")
+    logger.info(f"----> [stat] scale max: {max_size}")
+
+    # Refine min/max to reduce outlier effect
+    min_size = max(min_size, avg_3std_min_size)
+    max_size = min(max_size, avg_3std_max_size)
 
     if rule == "min":
-        object_area = np.min(areas)
+        object_size = min_size
     elif rule == "avg":
-        object_area = np.mean(areas)
-    max_area = np.max(areas)
-
-    tile_size = int(math.sqrt(object_area / object_tile_ratio))
-    tile_overlap = max_area / (tile_size**2)
+        object_size = avg_size
+
+    logger.info("[Adaptive tiling pararms]")
+    object_tile_ratio = tiling_parameters.object_tile_ratio
```

Comment:
This value should be set from the caller. What do you think? @sungmanc

Reply:
Ah... the tiling parameters are given....
Let's do either of: set from the caller, remove object_tile_ratio arg from this function.